site/oauth2-examples-azure.md

```diff
@@ -0,0 +1,237 @@
+# Use Azure Active Directory (Azure AD) as OAuth 2.0 server
+
+We are going to test 3 OAuth flows:
+1. Access management ui via a browser :ballot_box_with_check:
+2. Access management rest api :construction:
+3. Access AMQP protocol :construction:
+
+## Prerequisites to follow this guide
+
+* Have an account in https://portal.azure.com.
+* Docker
+* Openssl
+
+## Register your app
+
+When using **Azure AD as OAuth 2.0 server**, your client app (in our case RabbitMQ) needs a way to trust the security tokens issued to it by the **Microsoft identity platform**. The first step in establishing that trust is by **registering your app** with the identity platform in Azure AD.
+
+<g-emoji class="g-emoji" alias="blue_book" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4d8.png">📘</g-emoji> More details about App registration in Azure AD are available [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app)
+
+Once you have logged onto your account in [Azure Portal](https://portal.azure.com), go to **Azure Active Directory** (use the search bar if you are not able to easily find it).
```

Comment:
Uuse numbered steps for this procedure and add full stops at the end of each step.